pkg/operator/starter.go

```diff
@@ -106,8 +112,25 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 	}
 	versionRecorder.SetVersion("operator", os.Getenv("OPERATOR_IMAGE_VERSION"))
 
-	nodeProvider := encryptiondeployer.NewDeploymentNodeProvider(operatorclient.TargetNamespace, kubeInformersForNamespaces)
-	deployer, err := encryptiondeployer.NewRevisionLabelPodDeployer("revision", operatorclient.TargetNamespace, kubeInformersForNamespaces, resourceSyncController, kubeClient.CoreV1(), kubeClient.CoreV1(), nodeProvider)
+	openshiftNodeProvider := encryptiondeployer.NewDeploymentNodeProvider(operatorclient.TargetNamespace, kubeInformersForNamespaces)
+	openshiftDeployer, err := encryptiondeployer.NewRevisionLabelPodDeployer("revision", operatorclient.TargetNamespace, kubeInformersForNamespaces, resourceSyncController, kubeClient.CoreV1(), kubeClient.CoreV1(), openshiftNodeProvider)
+	if err != nil {
+		return err
+	}
+	openshiftDeployerAdapter := revisionpoddeployer.NewAlwaysEnabledDeployerAdapter(openshiftDeployer)
+
+	oauthNodeProvider := encryptiondeployer.NewDeploymentNodeProvider(oauthAPIServerTargetNamespace, kubeInformersForNamespaces)
+	// note: do not pass resourceSyncController - CAO is responsible for synchronization
+	oauthDeployer, err := encryptiondeployer.NewRevisionLabelPodDeployer("revision", oauthAPIServerTargetNamespace, kubeInformersForNamespaces, nil, kubeClient.CoreV1(), kubeClient.CoreV1(), oauthNodeProvider)
+	if err != nil {
+		return err
+	}
+	oauthDeployerAdapter := revisionpoddeployer.NewDisabledByPredicateDeployer(
+		encryptionprovider.IsOAuthEncryptionConfigManagedByThisOperator(
```

Comment:
move predicate out here to:
```Golang
weManageOauthConfig := encryptionprovider.IsOAuthEncryptionConfigManagedByThisOperator( ... )
deployer, err := revisionpoddeployer.NewUnionDeployer(
  deployer.AlwaysEnabled(openshiftDeployerAdapter),
  deployer.NewDisabledByPredicateDeployer(weManageOauthConfig, oauthDeployer),
)
```

enabled => disabled in the predicate deployer might make this easier to read?